test/resources/array-shims.ts

```diff
@@ -0,0 +1,30 @@
+import assert from "assert";
```

Comment:
I imagine these shims gets generated by TS now.

Reply:
I'm not sure what exactly you are referring to. Typescript does not (as far as I know) have any more expressive power than js, we still need to implement the flat and dedupe function (I'm getting type errors if I try to remove them).

Reply:
TS sometimes generates extra JS code when these sorts of methods are missing. For example:
```
const foo = {
    a: 1,
    b: 2,
};
const bar = { ...foo, c: 3 };
```

When targetting ES2017, would generate:
```
"use strict";
const foo = {
    a: 1,
    b: 2,
};
const bar = Object.assign(Object.assign({}, foo), { c: 3 });
```

But when targeting older ES5:
```
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var foo = {
    a: 1,
    b: 2,
};
var bar = __assign(__assign({}, foo), { c: 3 });
```

I just imagined it would do the same for `.flat()` but it seems it doesn't.